### PR TITLE
Refactor ops.rs

### PR DESCRIPTION
**Pull request by clarfonthey**
This refactors ops.rs into several different modules internally, as the file has gotten quite big. None of these modules are actually exported, but this should make maintaining it much easier. I've avoided the ambition of exporting the modules because they can more easily be rearranged after this commit goes through, even though it'd be cool to potentially export the modules in the future.

I've separated the creation of each file into a separate commit so that this is easier to read.

Redone version of #42269 with the movement of `RangeArgument` moved.

### CODE REVIEW

**Comment by kennytm** [comment]
Codegen failed.

```ini
[00:49:17] ---- [codegen-units] codegen-units/item-collection/function-as-argument.rs stdout ----
[00:49:17] 	
[00:49:17] These items should have been contained but were not:
[00:49:17] 
[00:49:17] TRANS_ITEM fn core::ops[0]::FnOnce[0]::call_once[0]<fn(char, f64), (char, f64)>
[00:49:17] TRANS_ITEM fn core::ops[0]::FnOnce[0]::call_once[0]<fn(u32, &str), (u32, &str)>
[00:49:17] 
[00:49:17] 
[00:49:17] 
[00:49:17] These items were contained but should not have been:
[00:49:17] 
[00:49:17] TRANS_ITEM fn core::ops[0]::function[0]::FnOnce[0]::call_once[0]<fn(char, f64), (char, f64)> @@ function_as_argument.cgu-0[Internal]
[00:49:17] TRANS_ITEM fn core::ops[0]::function[0]::FnOnce[0]::call_once[0]<fn(u32, &str), (u32, &str)> @@ function_as_argument.cgu-0[Internal]
[00:49:17] 
[00:49:17] 
[00:49:17] thread '[codegen-units] codegen-units/item-collection/function-as-argument.rs' panicked at 'explicit panic', /checkout/src/tools/compiletest/src/runtest.rs:1979
[00:49:17] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:49:17] 
[00:49:17] ---- [codegen-units] codegen-units/item-collection/trait-method-as-argument.rs stdout ----
[00:49:17] 	
[00:49:17] These items should have been contained but were not:
[00:49:17] 
[00:49:17] TRANS_ITEM fn core::ops[0]::FnMut[0]::call_mut[0]<fn(char) -> char, (char)>
[00:49:17] TRANS_ITEM fn core::ops[0]::FnMut[0]::call_mut[0]<fn(u32) -> u32, (u32)>
[00:49:17] TRANS_ITEM fn core::ops[0]::FnOnce[0]::call_once[0]<fn(char) -> char, (char)>
[00:49:17] TRANS_ITEM fn core::ops[0]::FnOnce[0]::call_once[0]<fn(u32) -> u32, (u32)>
[00:49:17] TRANS_ITEM fn core::ops[0]::Fn[0]::call[0]<fn(char) -> char, (char)>
[00:49:17] TRANS_ITEM fn core::ops[0]::Fn[0]::call[0]<fn(u32) -> u32, (u32)>
[00:49:17] 
[00:49:17] 
[00:49:17] 
[00:49:17] These items were contained but should not have been:
[00:49:17] 
[00:49:17] TRANS_ITEM fn core::ops[0]::function[0]::FnMut[0]::call_mut[0]<fn(char) -> char, (char)> @@ trait_method_as_argument.cgu-0[Internal]
[00:49:17] TRANS_ITEM fn core::ops[0]::function[0]::FnMut[0]::call_mut[0]<fn(u32) -> u32, (u32)> @@ trait_method_as_argument.cgu-0[Internal]
[00:49:17] TRANS_ITEM fn core::ops[0]::function[0]::FnOnce[0]::call_once[0]<fn(char) -> char, (char)> @@ trait_method_as_argument.cgu-0[Internal]
[00:49:17] TRANS_ITEM fn core::ops[0]::function[0]::FnOnce[0]::call_once[0]<fn(u32) -> u32, (u32)> @@ trait_method_as_argument.cgu-0[Internal]
[00:49:17] TRANS_ITEM fn core::ops[0]::function[0]::Fn[0]::call[0]<fn(char) -> char, (char)> @@ trait_method_as_argument.cgu-0[Internal]
[00:49:17] TRANS_ITEM fn core::ops[0]::function[0]::Fn[0]::call[0]<fn(u32) -> u32, (u32)> @@ trait_method_as_argument.cgu-0[Internal]
[00:49:17] 
[00:49:17] 
[00:49:17] thread '[codegen-units] codegen-units/item-collection/trait-method-as-argument.rs' panicked at 'explicit panic', /checkout/src/tools/compiletest/src/runtest.rs:1979
[00:49:17] 
[00:49:17] 
[00:49:17] failures:
[00:49:17]     [codegen-units] codegen-units/item-collection/function-as-argument.rs
[00:49:17]     [codegen-units] codegen-units/item-collection/trait-method-as-argument.rs
```

**Comment by clarcharr** [comment]
I updated the tests, so, it should work now.

**Comment by frewsxcv** [comment]
Latest Travis run has more codegen errors.

File: src/test/codegen-units/item-collection/trait-method-as-argument.rs [hunk 1]
```diff
@@ -40,27 +40,27 @@ fn take_foo_mut<T, F: FnMut(T) -> T>(mut f: F, arg: T) -> T {
 fn main() {
     //~ TRANS_ITEM fn trait_method_as_argument::take_foo_once[0]<u32, fn(u32) -> u32>
     //~ TRANS_ITEM fn trait_method_as_argument::{{impl}}[0]::foo[0]
-    //~ TRANS_ITEM fn core::ops[0]::FnOnce[0]::call_once[0]<fn(u32) -> u32, (u32)>
+    //~ TRANS_ITEM fn core::ops[0]::function[0]::FnOnce[0]::call_once[0]<fn(u32) -> u32, (u32)>
     take_foo_once(Trait::foo, 0u32);
 
-    //~ TRANS_ITEM fn trait_method_as_argument::take_foo_once[0]<char, fn(char) -> char>
+    //~ TRANS_ITEM fn trait_method_as_argument::take_foo_once[0]<char, fn(char) -> chr>
```
Comment (kennytm) [inline src/test/codegen-units/item-collection/trait-method-as-argument.rs]: Typo 😅

Comment (clarcharr) [inline src/test/codegen-units/item-collection/trait-method-as-argument.rs]: ><

**Comment by clarcharr** [comment]
I got rid of the typo, so, it should work now.

**Pull request merged**